packages/filebrowser/src/listing.ts

```diff
@@ -3603,7 +3603,7 @@ namespace Private {
       // Sort by name
       copy.sort(
         compare((a: Contents.IModel, b: Contents.IModel) => {
-          return b.name.localeCompare(a.name);
+          return b.name.localeCompare(a.name, navigator.language || navigator.userLanguage, {numeric: true, sensitivity: 'base'});
```

Comment:
I think that instead of `navigator.language || navigator.userLanguage` we should first try to use the language of the UI. You would need to pass `ITranslator` instance and access its [`languageCode`](https://jupyterlab.readthedocs.io/en/latest/api/interfaces/rendermime_interfaces.IRenderMime.ITranslator.html#languageCode) parameter. I am just not 100% sure if the language codes have the same BCP 47 format.

Reply:
The embarrassing part is that I’m not sure how to run the project from source. I tried starting with `npm install`, but it failed.

To make the change, I had to do some reverse engineering: I modified the compiled JavaScript code, then traced it back to the source.

Reply:
We use `yarn v3` rather than `npm`. Invoking just `jlpm` (a pinned version of `yarn` berry) should install the dependencies for you. See the full instructions here: https://jupyterlab.readthedocs.io/en/latest/developer/contributing.html#installing-jupyterlab

Reply:
The build on CI is failing with:

```
../filebrowser/src/listing.ts(3606,79): error TS2551: Property 'userLanguage' does not exist on type 'Navigator'. Did you mean 'language'?
```

Reply:
The `userLanguage` section was used for compatibility with older browsers, so it can be safely removed now that we are targeting modern browsers.